VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionComment.java

```diff
@@ -114,4 +114,26 @@ public boolean isAbuseFlagged() {
         return children;
     }
 
+    public IAuthorData getEndorserData() {
+        if (!endorsed) {
+            return null;
+        } else {
+            return new IAuthorData() {
```

Comment:
This could be reduced to a ternary conditional statement:

```` java
return !endorsed ? null : new IAuthorData() {
````

Also, you could create a lazily initialized singleton object for this.